src/lib/core/WeaveTLVWriter.cpp

```diff
@@ -878,7 +878,10 @@ WEAVE_ERROR TLVWriter::VPutStringF(uint64_t tag, const char *fmt, va_list ap)
     // no facilities for splitting the stream across multiple buffers,
     // just write however much fits in the current buffer.
     // assume conservative tag length at this time (8 bytes)
-    maxLen =  mRemainingLen - (1 + 8 + (1 << static_cast<uint8_t>(lenFieldSize)) + 1); // 1 : control byte, 8 : tag length, stringLen + 1 for null termination
+    size_t headerLen = (1 + 8 + (1 << static_cast<uint8_t>(lenFieldSize)) + 1); // 1 : control byte, 8 : tag length, stringLen + 1 for null termination
+    VerifyOrExit(mRemainingLen >= headerLen, err = WEAVE_ERROR_BUFFER_TOO_SMALL);
```

Comment:
Is WEAVE_ERROR_BUFFER_TOO_SMALL an accurate error to return? If this check is false, it means that there is not enough space left

Reply:
Yup, thats the right error